bugtool/cmd/configuration.go

```diff
@@ -0,0 +1,247 @@
+	c := BugtoolConfiguration{defaultCommands(confDir, ciliumDir, k8sPods)}
+	return &c, save(&c, path)
+}
+
+func defaultCommands(confDir string, ciliumDir string, k8sPods []string) []string {
+	var commands []string
+	// Not expecting all of the commands to be available
+	commands = []string{
+		// Host and misc
+		"ps",
+		"hostname",
+		"ip a",
+		"ip r",
+		"ip link",
+		"uname -a",
+		"dig",
+		"netstat",
+		"pidstat",
+		"arp",
+		"top -b -n 1",
+		"uptime",
+		"dmesg",
+		"bpftool map show",
+		"bpftool prog show",
+		// Versions
+		"docker version",
+		// Docker and Kubernetes logs from systemd
+		"journalctl -u cilium*",
+		"journalctl -u kubelet",
+		// iptables
+		"iptables-save",
+		"iptables -S",
+		"ip6tables -S",
+		"iptables -L -v",
+	}
+
+	// Commands that require variables and more configuration are added
+	// separately below
+	commands = append(commands, catCommands()...)
+	commands = append(commands, ethoolCommands()...)
+	commands = append(commands, copyConfigCommands(confDir, k8sPods)...)
+	commands = append(commands, copyCiliumInfoCommands(ciliumDir, k8sPods)...)
+	commands = append(commands, k8sCommands(commands, k8sPods)...)
+	return commands
+}
+
+func save(c *BugtoolConfiguration, path string) error {
+	f, err := os.Create(path)
+	if err != nil {
+		return fmt.Errorf("Failed to open file %s for writing: %s\n", path, err)
+	}
+	defer f.Close()
+
+	data, err := json.MarshalIndent(c, "", "\t")
+	if err != nil {
+		return fmt.Errorf("Cannot marshal config %s", err)
+	}
+	err = ioutil.WriteFile(path, data, 0644)
+	if err != nil {
+		return fmt.Errorf("Cannot write config %s", err)
+	}
+	return nil
+}
+
+func loadConfigFile(path string) (*BugtoolConfiguration, error) {
+	var content []byte
+	var err error
+	logrus.WithField(logfields.Path, path).Debug("Loading file")
+
+	content, err = ioutil.ReadFile(path)
+
+	if err != nil {
+		return nil, err
+	}
+
+	var c BugtoolConfiguration
+	err = json.Unmarshal(content, &c)
+	return &c, err
+}
+
+func catCommands() []string {
+	files := []string{
+		"/proc/sys/net/core/bpf_jit_enable",
+		"/proc/kallsyms",
+		"/etc/resolv.conf",
+		"/var/log/upstart/docker.log",
+		"/var/log/docker.log",
+		"/var/log/daemon.log",
+		"/var/log/messages",
+	}
+	// Only print the files that do exist to reduce number of errors in
+	// archive
+	commands := []string{}
+	for _, f := range files {
+		if _, err := os.Stat(f); os.IsNotExist(err) {
+			continue
+		}
+		commands = append(commands, fmt.Sprintf("cat %s", f))
+	}
+	// TODO: handle K8s case as well.
+	return commands
+}
+
+func copyConfigCommands(confDir string, k8sPods []string) []string {
+	commands := []string{}
+	type Location struct {
+		Src string
+		Dst string
+	}
+
+	locations := []Location{
+		{"/proc/config", fmt.Sprintf("%s/kernel-config", confDir)},
+		{"/proc/config.gz", fmt.Sprintf("%s/kernel-config.gz", confDir)},
+	}
+
+	if len(k8sPods) == 0 {
+		kernel, _ := execCommand("uname", "-r")
```

Comment:
Please add a comment describing what L142-L152 is doing

Reply:
Fixed